apps/core/scenes/bookingDetail/sectorDetails/segments/SegmentMap.js

```diff
@@ -6,50 +6,121 @@ import { StyleSheet } from '@kiwicom/universal-components';
+  +name: ?string,
+  +locationId: string,
+|};
+
+type CompareByType = {|
+  +locationId: string,
+|};
+
 const DEFAULT_PADDING = { top: 40, right: 40, bottom: 40, left: 40 };
 
-const mapCoordinates = memoize(
-  (segmentLocations: ?$PropertyType<BookingType, 'segmentLocations'>) => {
-    const coordinates = segmentLocations ?? [];
-    return coordinates.map(coordinate => ({
-      latitude: coordinate?.lat,
-      longitude: coordinate?.lng,
-    }));
-  },
-);
+const mapMarkerData = memoize((props: Props) => {
+  switch (props.data?.type) {
+    case 'BOOKING_ONE_WAY':
+      return mapSegmentsToMarkerData(props.data?.sector?.segments);
+    case 'BOOKING_RETURN':
+      return mapSegmentsToMarkerData(getReturnSegments(props));
+    case 'BOOKING_MULTICITY':
+      return mapSegmentsToMarkerData(getMulticitySegments(props));
+    default:
+      return null;
+  }
+});
+
+const getReturnSegments = (props: Props) => {
+  const outbound = props.data?.outbound?.segments ?? [];
+  const inbound = props.data?.inbound?.segments ?? [];
+  return [...outbound, ...inbound];
+};
 
-class SegmentMap extends React.Component<Props> {
+const getMulticitySegments = (props: Props) => {
+  const sectors = props.data?.sectors ?? [];
+  return sectors.reduce((acc, curr) => {
+    const segments = curr?.segments ?? [];
+    return [...acc, ...segments];
+  }, []);
+};
+
+const mapSegmentsToMarkerData = (segments: Segments) => {
+  if (segments == null) {
+    return [];
+  }
+  const bookingSegments = segments.reduce((acc, curr) => {
+    return [
+      ...acc,
+      {
+        ...curr?.departure?.stop?.coordinates,
+        name: curr?.departure?.stop?.city?.name,
+        locationId: curr?.departure?.stop?.locationId ?? '',
+      },
+      {
+        ...curr?.arrival?.stop?.coordinates,
+        name: curr?.arrival?.stop?.city?.name,
+        locationId: curr?.arrival?.stop?.locationId ?? '',
+      },
+    ];
+  }, []);
+
+  return uniqBy<MarkerData, CompareByType>(
+    prop<string, MarkerData>('locationId'),
+    bookingSegments,
+  );
+};
+
+export class SegmentMap extends React.Component<Props> {
   map: ?React.Ref<typeof MapView>;
 
   setRef = (ref: ?React.Ref<typeof MapView>) => {
     this.map = ref;
   };
 
-  getMappedCoordinates = () => {
-    return mapCoordinates(this.props.data?.segmentLocations);
-  };
+  getMarkerData = () => mapMarkerData(this.props);
 
   fit = () => {
     if (this.map != null) {
       // $FlowExpectedError: fitToCoordinates does exist on map, but flow is confused by .native|.web exports
-      this.map.fitToCoordinates(this.getMappedCoordinates(), {
+      this.map.fitToCoordinates(this.getMarkerData(), {
         edgePadding: DEFAULT_PADDING,
       });
     }
   };
 
   render() {
-    const coordinates = this.props.data?.segmentLocations ?? [];
-    const center = getCenter(this.getMappedCoordinates());
+    const locations = this.getMarkerData() ?? [];
+    const center = getCenter(this.getMarkerData());
```

Comment:
```suggestion
    const center = getCenter(locations);
```
To avoid invoking the function again.